pkg/scaffold/internal/machinery/scaffold_test.go

```diff
@@ -0,0 +1,453 @@
+						},
+					},
+				)).To(Succeed())
+				Expect(output.String()).To(Equal(fileContent))
+			})
+
+			It("should error if configured to do so", func() {
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							Path:           "filename",
+							IfExistsAction: file.Error,
+							TemplateBody:   fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("failed to create filename: file already exists"))
+				Expect(output.String()).To(BeEmpty())
+			})
+		})
+
+		Context("when the filesystem returns an error", func() {
+
+			It("should fail if fs.Exists failed", func() {
+				s = &scaffold{
+					fs: filesystem.NewMock(
+						filesystem.MockExistsError(testError),
+					),
+				}
+
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							TemplateBody: fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(testError.Error()))
+			})
+
+			It("should fail if fs.Create was unable to create the directory", func() {
+				s = &scaffold{
+					fs: filesystem.NewMock(
+						filesystem.MockCreateDirError(testError),
+					),
+				}
+
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							TemplateBody: fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(filesystem.IsCreateDirectoryError(err)).To(BeTrue())
+			})
+
+			It("should fail if fs.Create was unable to create the file", func() {
+				s = &scaffold{
+					fs: filesystem.NewMock(
+						filesystem.MockCreateFileError(testError),
+					),
+				}
+
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							TemplateBody: fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(filesystem.IsCreateFileError(err)).To(BeTrue())
+			})
+
+			It("should fail if fs.Create().Write was unable to write the file", func() {
+				s = &scaffold{
+					fs: filesystem.NewMock(
+						filesystem.MockWriteFileError(testError),
+					),
+				}
+
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							TemplateBody: fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(filesystem.IsWriteFileError(err)).To(BeTrue())
+			})
+
+			It("should fail if fs.Create().Write was unable to close the file", func() {
+				s = &scaffold{
+					fs: filesystem.NewMock(
+						filesystem.MockCloseFileError(testError),
+					),
+				}
+
+				err := s.Execute(
+					model.NewUniverse(),
+					fakeFile{
+						input: file.Input{
+							TemplateBody: fileContent,
+						},
+					},
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(filesystem.IsCloseFileError(err)).To(BeTrue())
+			})
+		})
+	})
+})
+
+// fakePlugin is used to mock a model.Plugin in order to test Scaffold
+type fakePlugin struct {
```

Comment:
Why we need the fakePlugin? Why not use the Plugin structure with mock data in the tests?

Reply:
`Plugin` is an interface. `fakePlugin` implements this interface. There is no Plugin structure nor data about it.

Reply:
Make sense you can close this one as solved.